java-frontend/src/main/java/org/sonar/java/AnalyzerMessage.java

```diff
@@ -139,29 +142,48 @@ public boolean isEmpty() {
       lastSyntaxToken.line(),
       lastSyntaxToken.column() + lastSyntaxToken.text().length()
     );
+    checkLocation(firstSyntaxToken, location);
+    return location;
+  }
+
+  private static AnalyzerMessage.TextSpan textSpanEndsWithTextBlock(SyntaxToken firstSyntaxToken, SyntaxToken lastSyntaxToken) {
```

Comment:
what about calling this method `textSpanBetween(SyntaxToken firstSyntaxToken, LiteralTree textBlockTree)` to cover this case explicitly instead?